### ISSUE TITLE
TPOT Heap problem 

### ISSUE THREAD

**Issue by firepro20** [dtag-dev-sec/tpotce#21]
Services not working, heap exceeding limit elasticsearch

![fgfgf](https://cloud.githubusercontent.com/assets/17837413/13763701/8a5df3c8-ea48-11e5-887d-04498e20d5be.png)

![fgfgf](https://cloud.githubusercontent.com/assets/17837413/13763379/1402daec-ea46-11e5-9fc4-8ddfbcea62bc.png)

There are no logs in data elk/log directory

**Comment by t3chn0m4g3** [comment]
@firepro20 Thanks for opening a new issue. Please login to T-Pot and run `sudo status.sh` and provide a screenshot of the output. Also need a screenshot of `htop` and the `/data/elk/log/elasticsearch.log`. What are the specs for your VM? At least 4GB RAM and 64GB diskspace?

**Comment by firepro20** [comment]
4GB RAM and 40GB of Space as instructed in the tutorial video

**Comment by t3chn0m4g3** [comment]
@firepro20 
Please ...
- [ ] provide a screenshot of the `sudo status.sh` output
- [ ] provide a screenshot of `htop`

**Comment by t3chn0m4g3** [comment]
@firepro20 Try giving it two cores.

**Comment by firepro20** [comment]
Gave it two cores and updated the screenshots

**Comment by t3chn0m4g3** [comment]
Let's get to work then ...

`sudo vi /etc/crontab` (you can use nano, too) and comment out the following line like this:
`# */5 * * * *   root 	check.sh`

`sudo vi /etc/init/elk.conf` and comment out the following line like this:
`#  rm -rf /data/elk/log/* || true`

Then `sudo reboot` and wait until you see ´sudo status.sh´ with the **FATAL** message again. Check you can now see logs entries in `/data/elk/log/elasticsearch.log` and upload it.

Just for curiosity, did you uncleanly shutdown your VM? If so, the elasticsearch index might got corrupted, preventing elasticsearch to start.

**Comment by firepro20** [comment]
I searched for the logs using ls in the data elk log directory but there is nothing. Normally to shutdown I use ACPI shutdown of Virtual Box

**Comment by t3chn0m4g3** [comment]
@firepro20 If there are still no logs even after you performing the steps mentioned above, then there is only one solution and that is reinstalling.
From what your are telling me it can only be the index, but without logs I have no idea how to help, sorry :cry:

**Comment by firepro20** [comment]
Ok I will reinstall again and setup everything. To avoid this mishap in the future and rule it out, how should I shutdown the TPOT and VM properly? I will give new machine 4GB RAM and 45GB HDD

**Comment by t3chn0m4g3** [comment]
@firepro20 

- `sudo poweroff`
- `sudo restart`
- `sudo shutdown`

Depending on how exposed your machine will be make sure to give it enough free diskspace. 64 GB is the minimum requirement.

**Comment by firepro20** [comment]
Ok will give it 64GB. And try again later this evening. On another note, when I get to Kibana on local host, is there a way to download logs in their pure format like I can download raw csv files of other visualizations? Sorry for posting this here just to close off the topic hopefully.

I would like to thankyou very much for your assistance thus far. You have been really helpful.

**Comment by t3chn0m4g3** [comment]
@firepro20 Usually you can find all the *raw* logs and honeypot data in `/data/`.

**Comment by firepro20** [comment]
Hi Marco. I did install the new Virtual Machine and set 64Gb and 4GB RAM. All was running fine. I decided to go to home folder to find the data folder, and I couldnt get in cause i didnt have permissions. 

So I gave myself admin access using sudo chown -R tsec /data/ and I got in. I found the logs.

As I was going through the /data/ path to find the desired logs, I refreshed 127.0.0.1:8080 Kibana app and the original problem occurred. So this might be the root cause of why the heap stack problem was occurring. So I tried to set the rights back to the root but did not work.

So basically how can I go in the data folder without running the above mentioned command and therefore ruining the system? 

With regards to powering down the machine, my machine is exposed to the outside world through port forwarding. I use sudo poweroff, and when window closes I close VirtualBox.

**Comment by t3chn0m4g3** [comment]
@firepro20 
`/data` needs to be owned by the user **tpot** with **760**. Just do a `chown -R tpot:tpot /data/*` followed by `chmod 760 -R /data/*` and `reboot`. If you want to access any file outside the permissions of **tsec** you need to be root. Just do a `sudo su -` then you can access `/data` and its files.

**Comment by firepro20** [comment]
I reinstalled everything to start from fresh. So now by default in its original state data folder belongs to tpot. If I decide to change access, I can give it back its original state by using the above chown command followed by chmod? 

Without touching any permissions, I can do the sudo su - and become root, view the necesary files and that way I do not mess with permissions and hopefully the heap stack problem does not occur

**Comment by t3chn0m4g3** [comment]
Yep.

If everything's working fine you can close the issue.
